### PR TITLE
Fix gas fees accounting in TransactionProcessor.Trace 

### PR DESCRIPTION
**Pull request by deffrian**
Fixes Closes Resolves #
#5680 

## Changes

- Now some of trace_* functions use Execute instead of trace.
- Do not refund in noValidation mode

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs [hunk 1]
```diff
@@ -241,7 +241,7 @@ public void Trace(Transaction transaction, BlockHeader block, ITxTracer txTracer
                 }
             }
 
-            UInt256 senderReservedGasPayment = noValidation ? UInt256.Zero : (ulong)gasLimit * effectiveGasPrice;
+            UInt256 senderReservedGasPayment = (ulong)gasLimit * effectiveGasPrice;
```
Comment (deffrian) [inline src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:244]: We subtract fee and then refund it. In Trace(noValidation = true) we subtract zero and then refund.

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs:244]: I'm quite sure we cannot do it for eth_call

File: src/Nethermind/Nethermind.Consensus/Tracing/Tracer.cs [hunk 2]
```diff
@@ -43,10 +45,12 @@ public Tracer(IWorldState stateProvider, IBlockchainProcessor blockProcessor, Pr
...
 
             blockTracer.EndBlockTrace();
-
-            return processedBlock;
         }
 
+        public void Trace(Block block, IBlockTracer tracer) => Process(block, tracer, _traceProcessor);
+
+        public void Execute(Block block, IBlockTracer tracer) => Process(block, tracer, _executeProcessor);
+
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Consensus/Tracing/Tracer.cs:53]: Why do we need separated processors?

File: src/Nethermind/Nethermind.Evm.Test/VirtualMachineTestsBase.cs [hunk 3]
```diff
@@ -118,7 +118,7 @@ protected TestAllTracerWithOutput Execute(long blockNumber, long gasLimit, byte[
         protected (Block block, Transaction transaction) PrepareTx(
             long blockNumber,
             long gasLimit,
-            byte[] code,
+            byte[] code = null,
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm.Test/VirtualMachineTestsBase.cs:121]: mark it nullable

File: src/Nethermind/Nethermind.Consensus/Tracing/ITracer.cs [hunk 4]
```diff
@@ -14,12 +14,18 @@ namespace Nethermind.Consensus.Tracing
...
-        /// <returns>Processed block</returns>
-        Block? Trace(Block block, IBlockTracer tracer);
+        void Trace(Block block, IBlockTracer tracer);
+
+        /// <summary>
+        /// Allows to trace and verify arbitrary constructed block. Subtracts gas from sender account
+        /// </summary>
+        /// <param name="block">Block to trace.</param>
+        /// <param name="tracer">Trace to act on block processing events.</param>
+        void Execute(Block block, IBlockTracer tracer);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Consensus/Tracing/ITracer.cs:28]: Can you add a write-up (maybe in <remarks> which JSON RPC uses which and why)

File: src/Nethermind/Nethermind.Consensus/Tracing/Tracer.cs (hunk 2 above)
Comment (deffrian) [reply src/Nethermind/Nethermind.Consensus/Tracing/Tracer.cs:53]: Wrote under previous comment

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**